airlock/business_logic.py

```diff
@@ -794,16 +794,32 @@ def release_files(self, release_request: ReleaseRequest, user: User):
         )
 
         for relpath, abspath in file_paths:
-            audit = AuditEvent.from_request(
-                request=release_request,
-                type=AuditEventType.REQUEST_FILE_RELEASE,
-                user=user,
-                path=relpath,
-            )
-            # Note: releasing the file updates its released_at and released by
-            # attributes, as an indication of intent to release. Actually uploading
-            # the file will be handled by the asychronous file uploader.
-            self._dal.release_file(release_request.id, relpath, user.username, audit)
+            # If the file has already been released, this is a re-release attempt due
+            # to a file upload issue with one or more files.
+            # Some files may have already been upoaded successfully, so for those
+            # we do nothing.
+            # Otherwise, we call release_file again to record the user that re-tried the
+            # upload attempt in the file metadata and audit log
+            request_file = release_request.get_request_file_from_output_path(relpath)
+            if not request_file.uploaded:
```

Comment:
Given that almost the whole loop body is inside the conditional, it might be slightly clearer to write this as a guard clause i.e:
```py
if request_file.uploaded:
    continue
```

Complete subjective stylistic preference though!